### PR TITLE
Add docker and apptainer image building on demand

### PR DESCRIPTION
**Pull request by mikemhenry**
This PR will let us build docker and apptainer (singulairty) images using github action's workflow dispatch feature

### CODE REVIEW

**Comment by mikemhenry** [comment]
Looks like the docker image builds fine but we run out of space when building the apptainer image, going to try a few hacks to fix this

File: .github/workflows/docker.yaml [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+# GitHub recommends pinning actions to a commit SHA.
+# To get a newer version, you will need to update the SHA.
+# You can also reference a tag or branch, but the action may change without warning.
+
+name: Create and publish a Docker image
+
+on:
+  push:
+    branches:
+      - feat/add_docker_and_apptainer
```
Comment (mikemhenry) [inline .github/workflows/docker.yaml:15]: I will remove this before we merge it in

**Review by ijpulidos — APPROVED** [review]
Looks good to me!

File: .github/workflows/docker.yaml [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+            -y -f >/dev/null 2>&1
+          sudo aptitude purge google-cloud-sdk -f -y >/dev/null 2>&1
+          sudo aptitude purge microsoft-edge-stable -f -y >/dev/null 2>&1 || true
+          sudo apt purge microsoft-edge-stable -f -y >/dev/null 2>&1 || true
+          sudo aptitude purge '~n ^mysql' -f -y >/dev/null 2>&1
+          sudo aptitude purge '~n ^php' -f -y >/dev/null 2>&1
+          sudo aptitude purge '~n ^dotnet' -f -y >/dev/null 2>&1
+          sudo apt-get autoremove -y >/dev/null 2>&1
+          sudo apt-get autoclean -y >/dev/null 2>&1
+          echo "some packages purged"
```
[the remark this answers is not in the document]
Comment (mikemhenry) [reply .github/workflows/docker.yaml:62]: This is actually running on the host, not inside the image. The image we base on is quite small, but the host has a TON of stuff, so I found that I ran out of space when building the singularity image after building the docker image.

[the remark this answers is not in the document]
Comment (mikemhenry) [reply .github/workflows/docker.yaml:62]: From here https://github.com/actions/runner-images#available-images and skimming that page, I don't think github plans on making any minimal images

File: .github/workflows/docker.yaml (hunk 1 above)
Comment (mikemhenry) [inline .github/workflows/docker.yaml:15]:
```suggestion
```

File: .github/workflows/docker.yaml (hunk 2 above)
[the remark this answers is not in the document]
Comment (ijpulidos) [reply .github/workflows/docker.yaml:59]: Oh ok, then this is as good as it gets! :) Thanks for checking that out.

[the remark this answers is not in the document]
Comment (mikemhenry) [reply .github/workflows/docker.yaml:59]: No problem! Thanks for giving me the idea to check!

**Pull request merged**